src/main/scala/org/constellation/lb/Manager.scala

```diff
@@ -231,3 +256,66 @@ class Manager(init: NonEmptyList[Addr], config: LoadbalancerConfig)(
       )
   }
 }
+
+object Manager {
```

Comment:
why those fns in object?

Reply:
They don't depend on the object's state so you don't really need  an instance to use them.

Reply:
I see, but they are like internal for the manager (mostly?), so imo they should be private fns inside

Reply:
filterKeys is generic enough to even be in a separate lib, the others can be private, but then, unit test them would be almost impossible.  I like to separate the pure functions from the stateful object in a separate module, but happy to move to private fn if is our standard.  
Another option is to make all the functions totally generic in `K` and `T` in `NonEmptyMap[K, Option[NonEmptyList[T]]]` and move to a totally different module.

Reply:
if you need internal functions to make tests, it's considered a "test smells" meaning most likely you're doing something wrong.

Reply:
I don't need the functions to make the tests,  is the other way around, I want to test those functions. Essentially the question is "do we want finer grained unit tests?"

Reply:
if so, they should be kinda external, not connected with the Manager itself

Reply:
I've ended up generalizing the functions and moving them to another module